### PR TITLE
feat(ums): introduce sign-out

### PR DESCRIPTION
**Pull request by reinos**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [our conventions](../CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
There is no `sign-out` function to remove the keys for an user.

Issue Number: N/A

## What is the new behavior?
A new function has been introduced `ums.signOut('john@doe.com')` to remove all tokens that belongs to the user and set, if needed, a new default tokens.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/api/core/componentStorage.ts [hunk 1]
```diff
@@ -41,6 +42,13 @@ export interface IStorageComponent {
    */
   setDefault(auth: string): void;
 
+  /**
+   * clear / Remove a token with a specific key
+   *
+   * @param key
+   */
+  removeTokens(key: string): void;
```
Comment (MishaJexia) [inline src/api/core/componentStorage.ts:50]: Is it Alias as a parameter? What does it mean key?

File: src/api/ums/umsModule.ts [hunk 2]
```diff
@@ -101,6 +101,15 @@ export class UMSModule<
     );
   }
 
+  /**
+   * Signs out an user by just removing the token that belongs to the user/alias
+   *
+   * @param email The email that functions as an alias for the tokenManager
+   */
+  public signOut(email: string): void {
```
Comment (MishaJexia) [inline src/api/ums/umsModule.ts:109]: Email might be not a good option. In sign user, we do not have email as an available parameter. I need to run getUser to get it. Maybe we can use some kind of a current user or something?

File: src/api/core/componentStorage.ts (hunk 1 above)
Comment (reinos) [reply src/api/core/componentStorage.ts:50]: Yes, will change it to `alias`! Good one!

File: src/api/ums/umsModule.ts (hunk 2 above)
Comment (reinos) [reply src/api/ums/umsModule.ts:109]: Same here, email is not correct. It should be `alias`. 

Taking this example 

```
ums.signIn({
  email: 'Elon@tesla.com',
  password: 'secret-password',
  default: true,
  alias: 'Elon Musk',
}).subscribe(user => {
  // run request with current user token
}, error => {
  console.log(error)
});
```

Both the `email` and `alias` are set as the alias for the tokens.

**Review by MishaJexia — APPROVED** [review]

**Review by veger — APPROVED** [review]

**Pull request merged**